pkg/instancetype/builder/builder.go

```diff
@@ -0,0 +1,146 @@
+package builder
+
+import (
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests/framework/cleanup"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+type InstancetypeOption func(*v1beta1.VirtualMachineInstancetype)
+
+func NewInstancetype(opts ...InstancetypeOption) *v1beta1.VirtualMachineInstancetype {
+	instancetype := &v1beta1.VirtualMachineInstancetype{}
+	for _, f := range opts {
+		f(instancetype)
+	}
+
+	if instancetype.Name == "" {
+		instancetype.GenerateName = "instancetype-"
```

Comment:
In other builders we populate `Name` with a random value instead. Might make sense to do that here as well? I think that makes it easier to use the builder in tests where the object is not sent to an API server.